### PR TITLE
Spring Boot migration - AAD related samples

### PR DESCRIPTION
**Pull request by saragluna**
Split PR #9395 into several PRs for the convenience of reviewing.

Modules included:

- azure-spring-boot-bom
- azure-spring-boot-sample-active-directory
- azure-spring-boot-sample-active-directory-backend
- azure-spring-boot-sample-active-directory-backend-v2
- azure-spring-boot-sample-active-directory-stateless

### CODE REVIEW

File: sdk/spring/azure-spring-boot-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        <artifactId>azure-spring-boot-starter</artifactId>
+        <version>2.2.5-beta.1</version> <!-- {x-version-update;com.microsoft.azure:azure-spring-boot-starter;current} -->
+      </dependency>
+      <dependency>
+        <groupId>com.microsoft.azure</groupId>
+        <artifactId>azure-active-directory-spring-boot-starter</artifactId>
+        <version>2.2.5-beta.1</version> <!-- {x-version-update;com.microsoft.azure:azure-active-directory-spring-boot-starter;current} -->
+      </dependency>
+      <!--
+      <dependency>
```
Comment (hemanttanwar) [inline sdk/spring/azure-spring-boot-bom/pom.xml:71]: Why this is commented out, if this is not needed, should we remove it ?

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-backend-v2/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+spring.security.oauth2.client.provider.azure-oauth-provider.token-uri=https://login.microsoftonline.com/{your-tenant-id}/oauth2/token
+spring.security.oauth2.client.provider.azure-oauth-provider.user-info-uri=https://login.microsoftonline.com/{your-tenant-id}/openid/userinfo
+spring.security.oauth2.client.provider.azure-oauth-provider.jwk-set-uri=https://login.microsoftonline.com/{your-tenant-id}/discovery/keys
+```
+---
+### Meet with `AADSTS240002: Input id_token cannot be used as 'urn:ietf:params:oauth:grant-type:jwt-bearer' grant` error.
+In Azure portal, app registration manifest page, configure `oauth2AllowImplicitFlow` in your application manifest to `true`. See [this issue](https://github.com/MicrosoftDocs/azure-docs/issues/8121#issuecomment-387090099) for details on this workaround.
+
+## Next steps
+## Contributing
```
Comment (hemanttanwar) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-backend-v2/README.md:96]: We should put standard text for contributing.

File: sdk/spring/azure-spring-boot-bom/pom.xml (hunk 1 above)
Comment (saragluna) [reply sdk/spring/azure-spring-boot-bom/pom.xml:71]: The commented out parts are in following PRs.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-backend-v2/README.md (hunk 2 above)
Comment (saragluna) [reply sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-backend-v2/README.md:96]: Should we link to this [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-java/blob/master/CONTRIBUTING.md)?

Comment (hemanttanwar) [reply sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-backend-v2/README.md:96]: This contributing guide is very specific to azure-sdk-for-java and its related tools. You probably want a guide which talks about how user can contribute to something which is specific to azure-spring-*

File: eng/jacoco-test-coverage/pom.xml [hunk 3]
```diff
@@ -187,6 +187,11 @@
       <artifactId>azure-sdk-template</artifactId>
       <version>1.0.4-beta.13</version> <!-- {x-version-update;com.azure:azure-sdk-template;current} -->
     </dependency>
+    <dependency>
```
[the remark this answers is not in the document]
Comment (saragluna) [reply eng/jacoco-test-coverage/pom.xml:190]: @JimSuplizio Thanks for the clarification. I'll define the versions of Spring modules in version_client.txt then. We'd like to use track 2 pipelines for Spring modules, but only with the group id `azure.microsoft.com` for as long as one more release for them.

**Comment by mitchdenny** [comment]
I agree with @JimSuplizio on BOMs. We should not create a BOM with the intention of referencing it as an internal dependency (that is, our Spring Boot libraries depend on it). If you choose to create a BOM then it will need to go in ```sdk/boms/[bom-name]``` but not be referenced anywhere else within our repository.

**Comment by saragluna** [comment]
@mitchdenny @JimSuplizio We're going to apply such changes, moving spring-bom to sdk/boms folder and removing dependencyManagement sections.

**Comment by saragluna** [comment]
@hemanttanwar @JimSuplizio Please help approve this PR.

**Comment by saragluna** [comment]
> @saragluna I'm going to pre-approve this however the spring BOM file should not be using the current or dependency tags (external dependency is okay) of the spring libraries. The BOM files versions should get updated before it's released which is supposed to be much less frequently than the library releases.

Thanks for pointing it out, I've removed all current tags from the BOM file.

**Pull request merged**